gremlin-go/driver/graphBinary.go

```diff
@@ -583,6 +578,28 @@ func pathReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return p, nil
 }
 
+// Format: Same as List.
+// Mostly similar to listWriter and listReader with small changes
+func setWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	slice, err := value.(Set).SetToSlice()
+	if err != nil {
+		return nil, err
+	}
+	b, err := listWriter(slice, buffer, typeSerializer)
+	if err != nil {
+		return nil, err
+	}
+	return b, err
+}
+
+func setReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) (interface{}, error) {
+	slice, err := listReader(buffer, typeSerializer)
+	if err != nil {
+		return nil, err
+	}
+	return new(SimpleSet).Add(slice.([]interface{})...), nil
```

Comment:
Could you just return` NewSimpleSet(slice.([]interface{})...), nil?` instead of duplicating that functions effort here?